packages/ui-library/src/elements/popover/index.js

```diff
@@ -0,0 +1,176 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import { XIcon } from "@heroicons/react/outline";
+import React, { createContext, forwardRef, useCallback, useContext } from "react";
+import { isArray, noop } from "lodash";
+
+const PopoverContext = createContext( { handleDismiss: noop  } );
+
+const positionClassNameMap = {
+	top: "yst-popover--top",
+	topLeft: "yst-popover--top-left",
+	topRight: "yst-popover--top-right",
+	right: "yst-popover--right",
+	bottom: "yst-popover--bottom",
+	left: "yst-popover--left",
+	bottomLeft: "yst-popover--bottom-left",
+	bottomRight: "yst-popover--bottom-right",
+};
+
+/**
+ * @returns {Object} The popover context.
+ */
+export const usePopoverContext = () => useContext( PopoverContext );
+
+/**
+ * @param {string} dismissScreenReaderLabel The screen reader label for the dismiss button.
+ * @param {string} [className] The additional class name.
+ * @returns {JSX.Element} The close button.
+ */
+const CloseButton = ( {
+	dismissScreenReaderLabel,
+} ) => {
+	const { handleDismiss } = usePopoverContext();
+
+	return (
+		<div className="yst-flex-shrink-0 yst-flex yst-self-start">
+			<button
+				type="button"
+				onClick={ handleDismiss }
+				className="yst-bg-transparent yst-rounded-md yst-inline-flex yst-text-slate-400 hover:yst-text-slate-500 focus:yst-outline-none focus:yst-ring-2 focus:yst-ring-offset-2 focus:yst-ring-primary-500"
+			>
+				<span className="yst-sr-only">{ dismissScreenReaderLabel }</span>
+				<XIcon className="yst-h-5 yst-w-5" />
+			</button>
+		</div>
+	);
+};
+
+CloseButton.propTypes = {
+	dismissScreenReaderLabel: PropTypes.string.isRequired,
+};
+
+/**
+ * @param {string|string[]} content The popover content.
+ * @param {string } id The id of the content for accessibility.
+ * @param {string} [className] The additional class name.
+ * @returns {JSX.Element} The content.
+ */
+const Content = ( {
+	content,
+	id,
+	className = "",
+} ) => {
+	return isArray( content ) ? (
```

Comment:
This all is pretty specific. I would definitely not expect a UL list as default content for a popover!

What makes this the Content?
Perhaps you actually mean `List` and `Paragraph` or something instead of combining them as `Content`?